cachi2/core/models/sbom.py

```diff
@@ -395,27 +482,50 @@ class SPDXSbom(pydantic.BaseModel):
 
     creationInfo: SPDXCreationInfo
     packages: list[SPDXPackage] = []
+    files: list[SPDXFile] = []
     relationships: list[SPDXRelation] = []
 
+    def __hash__(self) -> int:
+        return hash(
+            hash(self.name + self.documentNamespace)
+            + hash(SPDXCreationInfo)
+            + sum(hash(p) for p in self.packages)
+            + sum(hash(f) for f in self.files)
+            + sum(hash(r) for r in self.relationships)
+        )
+
+    @classmethod
+    def from_file(cls, path: Path) -> "SPDXSbom":
+        """Consume a SPDX json directly from a file."""
+        return cls.model_validate_json(path.read_text())
+
     @staticmethod
     def deduplicate_spdx_packages(items: Iterable[SPDXPackage]) -> list[SPDXPackage]:
         """Deduplicate SPDX packages and merge external references.
 
         If package with same name and version is found multiple times in the list,
         merge external references of all the packages into one package.
         """
+        # NOTE: keeping this implementation mostly intact for historical reasons.
```

Comment:
Can you elaborate, this implementation was done within this very PR so indirectly referring to it as "inferior" and only keeping it intact for historical reasons doesn't seem very convincing in general.